src/main/java/db2/onlineshop/service/impl/BasicMovieInfo.java

```diff
@@ -0,0 +1,52 @@
+package db2.onlineshop.service.impl;
+
+import db2.onlineshop.entity.Movie;
+import db2.onlineshop.service.MovieInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BasicMovieInfo implements MovieInfo {
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    // todo: add enrich obj
+    private MovieInfo genreService;
```

Comment:
Spring ... Provider 
List < MovieEnricher > enricherList;
public void enrich(Movie result) {
   for(MovieEnricher enricher : enricherList){
        enricher.enrich(movie);
   }
}